### ISSUE TITLE
erratic driving

### ISSUE THREAD

**Issue by Blainer75** [Courseplay/courseplay#1145]
my grain trucks drive erratically swerving back and forth when following course after unloading from harvester. have tried making different routes but they always seem to swerve sometimes causeing crashed into trees and fences. am i doing something wrong.

**Comment by Mopar** [comment]
I have experienced this also. From other posts here and from some experimentation, here is what I know/found out.

1. Any drops in framerate can cause issues with CP and how it works/follows the course. Some fixes/work-arounds are to remove mods that tend to cause lag (like the autosave mod for example). Other sources of *potential* lag from mods are ones that spam error msgs in the log. If you have any like that, try to fix/replace/remove the mod causing that issue. Another way is to lower your graphics settings to see if that helps.

2. Speed related. I have noticed that some equipment tend to slide very easily (either intentional or improper creating of the mod) and that can cause issues. I have found speeds around 30~35 MPH (not sure what that would be in KPH) seem to be the least problematic.

As for speed related issues, another possible fix on CP's side would be to have it scan 2~3 waypoints ahead to adjust its speed/steering accordingly. Currently it *seems* like it only looks for the next waypoint and uses that for steering and the speed from the current/previous waypoint.
